library/spdm_responder_lib/libspdm_rsp_challenge_auth.c

```diff
@@ -79,11 +79,15 @@ libspdm_return_t libspdm_get_response_challenge_auth(void *context,
                                                SPDM_ERROR_CODE_INVALID_REQUEST, 0,
                                                response_size, response);
     }
-    if (!libspdm_is_capabilities_flag_supported(spdm_context, false, 0,
-                                                SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP) &&
-        spdm_request->header.param2 > 0) {
-        return libspdm_generate_error_response (spdm_context, SPDM_ERROR_CODE_INVALID_REQUEST,
-                                                SPDM_CHALLENGE, response_size, response);
+    if (spdm_request->header.param2 > 0) {
+        if (!libspdm_is_capabilities_flag_supported(
+                spdm_context, false, 0,
+                SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP) ||
+            (spdm_context->connection_info.algorithm.measurement_spec == 0) ||
+            (spdm_context->connection_info.algorithm.measurement_hash_algo == 0) ) {
```

Comment:
Aren't these equivalent? If Requester sets `measurement_spec` to zero then `measurement_hash_algo` will be zero, right?

Reply:
It should be at this time.
But I do not know what it will be in the future or in some corner case. This is more a defensive way.